src/topology/local_at_target.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.maps
+import topology.sets.opens
+import topology.metric_space.basic
+
+/-!
+# Properties of maps that are local at the target.
+
+We show that the following properties of continuous maps are local at the target :
+- `inducing`
+- `embedding`
+- `open_embedding`
+- `closed_embedding`
+
+-/
+
+open topological_space
+
+variables {α β : Type*} [topological_space α] [topological_space β] (f : α → β) (h : continuous f)
+variables {s : set β} {ι : Type*} (U : ι → opens β) (hU : supr U = ⊤)
+
+lemma continuous_at.restrict_preimage (s : set β) (x : f ⁻¹' s)
+  (h : continuous_at f x) : continuous_at (s.restrict_preimage f) x :=
+begin
+  intros U hU,
+  obtain ⟨_, e, ⟨V, hV, rfl⟩, hx⟩ := mem_nhds_iff.mp hU,
+  have := h (is_open.mem_nhds hV hx),
+  rw [filter.mem_map, mem_nhds_iff] at this ⊢,
+  obtain ⟨W, e', hW, hx'⟩ := this,
+  use coe ⁻¹' W,
+  refine ⟨_, continuous_subtype_coe.1 _ hW, hx'⟩,
+  intros y hy,
+  apply e,
+  exact e' hy
+end
```

Comment:
Can you add the following lemmas (in appropriate files):
```lean
lemma continuous_at_cod_restrict {t : set β} (h1 : ∀ x, f x ∈ t) {x : α} :
  continuous_at (cod_restrict f t h1) x ↔ continuous_at f x :=
by simp_rw [inducing_coe.continuous_at_iff, function.comp, coe_cod_restrict_apply]

alias continuous_at_cod_restrict ↔ _ continuous_at.cod_restrict

lemma continuous_at.restrict {s : set α} {t : set β} (h1 : maps_to f s t) {x : s}
  (h2 : continuous_at f x) : continuous_at (h1.restrict f s t) x :=
(h2.comp continuous_at_subtype_coe).cod_restrict _
```
These generalize this result, and make this proof just `h.restrict _`.